AMF pod crash when Mobile data disabled and re-enabled in UE

ashithacdac opened the pull request:
Issue: We have successfully connected the SD core with the OAI-gNB (usrp B210) and real UE. However, the pod crashes consistently when we disable and re-enable mobile data.

Troubleshoot: We identified the root cause of the pod crash: the Pdusessionresourcesetupresponse message contains both PDUSessionResourceSetupListSures and PDUSessionResourceFailedToSetupListSures IE's. While PDUSessionResourceSetupListSures contains the expected pduSessionId matching the one sent in the PDUSessionResourceSetupRequest, PDUSessionResourceFailedToSetupListSures contains a pduSessionId of 0, leading to the pod crash.

thakurajayL commented:
Thank You for providing fix. Could you please provide pcap as well. May help understand RCA better.

ashithacdac commented:
Yeah Sure.
[Mobile_Data_Toggle.zip](https://github.com/user-attachments/files/16792351/Mobile_Data_Toggle.zip)

gab-arrobo commented on ngap/handler.go:
```diff
@@ -1717,22 +1717,24 @@ func HandlePDUSessionResourceSetupResponse(ran *context.AmfRan, message *ngapTyp
+					transfer := item.PDUSessionResourceSetupUnsuccessfulTransfer
+					smContext, ok := amfUe.SmContextFindByPDUSessionID(pduSessionID)
+					if !ok {
+						ranUe.Log.Errorf("SmContext[PDU Session ID:%d] not found", pduSessionID)
```
Agree with @thakurajayL that `continue` is the appropriate way to handle this case.

ashithacdac commented:
By mistake, I have deleted the earlier branch. Incorporated the requested changes in the PR. Please check the same.
https://github.com/omec-project/amf/pull/291
